runtime/src/main/java/org/corfudb/runtime/GarbageInformer.java

```diff
@@ -22,6 +55,26 @@ public synchronized void add(UUID streamId, long detectorAddress, SMREntry.SMREn
         }
     }
 
+    public void gc() {
+        garbageQueue.addAll(addressToGarbage.entrySet());
+        addressToGarbage.clear();
+    }
+
+    public void garbageInformationProcessor() {
+        try {
+            while (true) {
+                Map.Entry<Long, ISMRGarbageInfo> garbage = garbageQueue.peek();
```

Comment:
Peek could return null.

Reply:
Good catch! It is changed to blocking take().